src/source_sampling.cpp

```diff
@@ -74,6 +74,27 @@ pyne::Sampler::Sampler(std::string filename,
   setup();
 }
 
+pyne::Sampler::Sampler(std::string filename,
+                 std::map<std::string, std::string> tag_names,
+                 std::vector<double> e_bounds, 
+                 int mode)
+  : filename(filename),
+    e_bounds(e_bounds) {
+  // determine the bias_mode
+  if (mode == 0){
+    bias_mode = ANALOG; 
+  } else if (mode == 1) {
+    bias_mode = UNIFORM;
+  } else if (mode == 2) {
+    bias_mode = USER;
+  }
+
+  // find out the filename, src_tag_name and bias_tag_name
+  src_tag_name = tag_names["src_tag_name"];
+  bias_tag_name = tag_names["bias_tag_name"];
```

Comment:
in general, don't we need to check for the existence of this tag?  It might not always exist.

Reply:
The input check is added in the update.